### PR TITLE
Priority headers should be set by the application

### PR DESCRIPTION
**Pull request by valenting**
Instead of setting the priority headers in neqo, the headers should be provided by the application.

See related https://phabricator.services.mozilla.com/D201265

### CODE REVIEW

File: neqo-http3/tests/priority.rs [hunk 1]
```diff
@@ -68,7 +68,7 @@ fn priority_update() {
             Instant::now(),
             "GET",
             &("https", "something.com", "/"),
-            &[],
+            &[Header::new("priority", "u=4,i")],
```
Comment (larseggert) [inline neqo-http3/tests/priority.rs:71]: Is the test verifying that this gets used somewhere?

File: neqo-http3/tests/priority.rs [hunk 2]
```diff
@@ -129,7 +129,7 @@ fn priority_update_dont_send_for_cancelled_stream() {
             Instant::now(),
             "GET",
             &("https", "something.com", "/"),
-            &[],
+            &[Header::new("priority", "u=5")],
```
Comment (larseggert) [inline neqo-http3/tests/priority.rs:132]: Ditto.

File: neqo-http3/tests/priority.rs (hunk 1 above)
Comment (valenting) [reply neqo-http3/tests/priority.rs:71]: Yes, line 96 in expected_headers.

File: neqo-http3/tests/priority.rs (hunk 2 above)
Comment (valenting) [reply neqo-http3/tests/priority.rs:132]: This test doesn't check the headers as far as I can tell, but I put them here for consistency.

**Comment by larseggert** [comment]
LGTM, but I'd like to see another review, since the related Fx parts are new to me.

**Review by KershawChang — APPROVED** [review]
Looks good.

**Pull request merged**